packages/dev/core/src/Rendering/IBLShadows/iblShadowsVoxelRenderer.ts

```diff
@@ -583,29 +605,26 @@ export class _IblShadowsVoxelRenderer {
 
     /**
      * Renders voxel grid of scene for IBL shadows
-     * @param excludedMeshes
+     * @param includedMeshes
      */
-    public updateVoxelGrid(excludedMeshes: number[]) {
+    public updateVoxelGrid(includedMeshes: Mesh[]) {
         this._stopVoxelization();
 
         this._voxelizationInProgress = true;
 
         if (this._triPlanarVoxelization) {
-            this._addRTsForRender(this._voxelMrtsXaxis, excludedMeshes, 0);
-            this._addRTsForRender(this._voxelMrtsYaxis, excludedMeshes, 1);
-            this._addRTsForRender(this._voxelMrtsZaxis, excludedMeshes, 2);
+            this._addRTsForRender(this._voxelMrtsXaxis, includedMeshes, 0);
+            this._addRTsForRender(this._voxelMrtsYaxis, includedMeshes, 1);
+            this._addRTsForRender(this._voxelMrtsZaxis, includedMeshes, 2);
         } else {
-            this._addRTsForRender(this._voxelMrtsZaxis, excludedMeshes, 2);
+            this._addRTsForRender(this._voxelMrtsZaxis, includedMeshes, 2);
         }
-
-        // Add the slab debug RT if needed.
         if (this._voxelDebugEnabled) {
-            this._removeVoxelRTs([this._voxelSlabDebugRT]);
-            this._addRTsForRender([this._voxelSlabDebugRT], [], this._voxelDebugAxis, 1, true);
+            this._addRTsForRender([this._voxelSlabDebugRT], includedMeshes, this._voxelDebugAxis, 1, true);
         }
 
         (this as any).boundVoxelGridRenderFn = this._renderVoxelGrid.bind(this);
-        this._scene.onAfterRenderTargetsRenderObservable.add((this as any).boundVoxelGridRenderFn);
+        this._scene.onAfterRenderTargetsRenderObservable.addOnce((this as any).boundVoxelGridRenderFn);

```

Comment:
Please don't use the "any" type except if strictly necessary: here you can add a `_boundVoxelGridRenderFn` property to the class.

Reply:
Done